### PR TITLE
Waveshaper symmetry

### LINKED ISSUE #2208: Asymmetric operation mode for waveshaper
I got the idea after watching this video:

https://www.youtube.com/watch?v=nj3st4lmKKo

The guy says about importance of asymmetrical waveshaping in sound design. I feel our great Waveshaper plugin could benefeit from this wisdom.

What do you think?

### PR DESCRIPTION
**Pull request by zonkmachine**
closes #2208

waveshapersymmetry

Symmetry. Add +/-0.5 DC component to the input.

### CODE REVIEW

File: plugins/waveshaper/waveshaper.cpp [hunk 1]
```diff
@@ -84,22 +84,30 @@ bool waveShaperEffect::processAudioBuffer( sampleFrame * _buf,
...
+	float *symmetryPtr = symmetryBuffer ? &( symmetryBuffer->values()[ 0 ] ) : &symmetry;
 
 	for( fpp_t f = 0; f < _frames; ++f )
 	{
 		float s[2] = { _buf[f][0], _buf[f][1] };
 
+// apply symmetry
+		s[0] += *symmetryPtr;
+		s[1] += *symmetryPtr;
+
```
Comment (zonkmachine) [inline plugins/waveshaper/waveshaper.cpp]: Pre or post gain?

**Comment by Umcaruje** [comment]
@unfa do you need some help with building and testing this PR? You were the OP of the issue, and I'm not comfortable merging this as I'm not exactly familiar with waveshapers and how they work :)

It seems like a nice addition, but it would be nice to see that it was done properly.

**Comment by zonkmachine** [comment]
> it would be nice to see that it was done properly.

:+1:  I'm not really into the waveshaper plugin so I would need feedback from someone who is.

**Comment by zonkmachine** [comment]
Resolved conflicts from gui redesign.

@unfa Can you test this? It was your idea and you have the bigger picture.

Should the symmetry function be applied before or after the input gain?
Is the amount of symmetry good?
Knob placement?

**Comment by unfa** [comment]
@zonkmachine - sorry for not providing any feedback, I totally missed this one. Is it still relevant for me to check this?

**Comment by zonkmachine** [comment]
This PR was tested by @unfa (thanks!) and was concluded to have some issues because of the dc offset on the output which breaks some plugins like the Equalizer and the Spectrum Analyzer.
It will need either to have the dc offset removed on the output or maybe some other method altogether. I won't have time to work on this right now, with 'right now' being flexible and generous in nature. Closing as abandoned...

**Pull request closed without merging**